sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/batch/BulkExecutorTest.java

```diff
@@ -234,6 +239,337 @@ public void executeBulk_OnGoneFailure() throws InterruptedException {
 
 
 
+    // Makes sure ordering is kept with normal flow of bulk
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void executeBulk_preserveOrdering() throws InterruptedException {
+        int totalRequest = 100;
+        this.container = createContainer(database);
+
+        List<CosmosItemOperation> cosmosItemOperations = new ArrayList<>();
+        String duplicatePK = UUID.randomUUID().toString();
+        String id = UUID.randomUUID().toString();
+        for (int i = 0; i < totalRequest; i++) {
+            if (i == 0) {
+                BatchTestBase.EventDoc eventDoc = new BatchTestBase.EventDoc(id, 2, 4, "type1",
+                    duplicatePK);
+                cosmosItemOperations.add(CosmosBulkOperations.getCreateItemOperation(eventDoc,
+                    new PartitionKey(duplicatePK)));
+            } else {
+                cosmosItemOperations.add(CosmosBulkOperations.getPatchItemOperation(id,
+                    new PartitionKey(duplicatePK),
+                    CosmosPatchOperations.create().replace("/type", "updated" + i)));
+            }
+        }
+
+        CosmosItemOperation[] itemOperationsArray =
+            new CosmosItemOperation[cosmosItemOperations.size()];
+        cosmosItemOperations.toArray(itemOperationsArray);
+        CosmosBulkExecutionOptions cosmosBulkExecutionOptions = new CosmosBulkExecutionOptions();
+        ImplementationBridgeHelpers.CosmosBulkExecutionOptionsHelper
+            .getCosmosBulkExecutionOptionsAccessor()
+            .setPreserveOrdering(cosmosBulkExecutionOptions, true);
+
+        Flux<CosmosItemOperation> inputFlux = Flux
+            .fromArray(itemOperationsArray)
+            .delayElements(Duration.ofMillis(100));
+        final BulkExecutorTemp<BulkExecutorTest> executor = new BulkExecutorTemp<>(
+            container,
+            inputFlux,
+            cosmosBulkExecutionOptions);
+
+
+        try {
```

Comment:
The `try` block can be placed further ahead.

Reply:
Put the try block lower